### PR TITLE
HPCC-27780 Fix uninitialized ESPenums error while copying queries

### PR DESCRIPTION
**Pull request by afishbeck**
This was a regression in 8.6.26.

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/scm/ws_workunits_struct.ecm [hunk 1]
```diff
@@ -528,8 +528,8 @@ ESPStruct WUFileOption
     int SlaveNumber(1); //ThorSlaveLog: optional
     int64 SizeLimit(0);
     [min_ver("1.90"), nil_remove] unsigned MaxLogRecords;
-    [min_ver("1.90"), nil_remove] ESPenum LogSelectColumnMode LogSelectColumnMode;
-    [min_ver("1.90"), nil_remove] ESPenum LogAccessLogFormat LogFormat;
+    [min_ver("1.90"), nil_remove] ESPenum LogSelectColumnMode LogSelectColumnMode(0);
```
Comment (ghalliday) [inline esp/scm/ws_workunits_struct.ecm:531]: Are these correct?  Shouldn't this be using the associated enumeration text e.g., "DEFAULT" (and say "XML" for LogFomat?)

**Review by ghalliday — COMMENTED** [review]
@afishbeck one question.

Also, looking back at https://github.com/hpcc-systems/HPCC-Platform/pull/15992/files I don't think it was really correct to reuse that WUFileOption structure.  I think it it something similar, but different.  For instance SlaveNumber and SizeLimit are now specified in two places.

**Comment by wangkx** [comment]
> Also, looking back at https://github.com/hpcc-systems/HPCC-Platform/pull/15992/files I don't think it was really correct to reuse that WUFileOption structure. I think it it something similar, but different. For instance SlaveNumber and SizeLimit are now specified in two places.
@ghalliday the WULogFileRequest has many individual parameters (SlaveNumber, SizeLimit, etc). To pass them into some CPP functions, CPP functions have to use many (more and more) input parameters. I was thinking to replace the individual parameters using the WUFileOption structure. In https://github.com/hpcc-systems/HPCC-Platform/pull/15992, the WUFileOption is used only for WU Component Log file. I am thinking to replace the individual parameters for other WU files gradually.

**Review by wangkx — APPROVED** [review]
Looks fine for me

File: esp/scm/ws_workunits_struct.ecm (hunk 1 above)
Comment (ghalliday) [reply esp/scm/ws_workunits_struct.ecm:531]: Would 1 be a more appropriate default?

**Review by ghalliday — APPROVED** [review]
@afishbeck explained why it was different.  Approved, but one question about if the default is correct.

**Comment by afishbeck** [comment]
Updated the default for LogSelectColumnMode to the one named "DEFAULT" which makes a bit more sense.

**Review by richardkchapman — APPROVED** [review]

**Pull request merged**